prometheus_client/exposition.py

```diff
@@ -81,35 +82,40 @@ def generate_latest(registry=core.REGISTRY):
     return ''.join(output).encode('utf-8')
 
 
-class MetricsHandler(BaseHTTPRequestHandler):
-    """HTTP handler that gives metrics from ``core.REGISTRY``."""
-
-    def do_GET(self):
-        registry = core.REGISTRY
-        params = parse_qs(urlparse(self.path).query)
-        if 'name[]' in params:
-            registry = registry.restricted_registry(params['name[]'])
-        try:
-            output = generate_latest(registry)
-        except:
-            self.send_error(500, 'error generating metric output')
-            raise
-        self.send_response(200)
-        self.send_header('Content-Type', CONTENT_TYPE_LATEST)
-        self.end_headers()
-        self.wfile.write(output)
-
-    def log_message(self, format, *args):
-        """Log nothing."""
+def factory_MetricsHandler(registry=core.REGISTRY):
```

Comment:
This name doesn't feel right. Do you know if there's conventions around this?

Reply:
I had a fast glance and didn't found anything.

I didn't try that, but it could work. Do you like it?

```
class MetricsHandler:
  ...old class stuff..
  @staticmethod
  def factory(registry):
     MH = type('MetricsHandler', (MetricsHandler,), {'registry': registry})
     return MetricsHandler
```

Reply:
That would work I think.

Reply:
Ok, here in Italy it's holiday. I'll give it a try asap. Thanks for all your support!